### PR TITLE
Pong

### PR DESCRIPTION
**Pull request by vanaacken**

### CODE REVIEW

**Comment by K1ngmar** [comment]
Build error:
```bash
backend     | [Nest] 39  - 03/30/2022, 12:58:18 PM   ERROR [ExceptionHandler] OAuth2Strategy requires a clientID option
backend     | TypeError: OAuth2Strategy requires a clientID option
backend     |     at GithubStrategy.OAuth2Strategy (/app/node_modules/passport-oauth2/lib/strategy.js:87:34)
backend     |     at new Strategy (/app/node_modules/passport-github2/lib/strategy.js:62:18)
backend     |     at new MixinStrategy (/app/node_modules/@nestjs/passport/dist/passport/passport.strategy.js:32:13)
backend     |     at new GithubStrategy (/app/src/auth/oauth/github.guard.ts:14:5)
backend     |     at Injector.instantiateClass (/app/node_modules/@nestjs/core/injector/injector.js:301:19)
backend     |     at callback (/app/node_modules/@nestjs/core/injector/injector.js:48:41)
backend     |     at processTicksAndRejections (node:internal/process/task_queues:96:5)
backend     |     at Injector.resolveConstructorParams (/app/node_modules/@nestjs/core/injector/injector.js:124:24)
backend     |     at Injector.loadInstance (/app/node_modules/@nestjs/core/injector/injector.js:52:9)
backend     |     at Injector.loadProvider (/app/node_modules/@nestjs/core/injector/injector.js:74:9)
```

File: src/backend/app/src/pong/pong.game.ts [hunk 1]
```diff
@@ -89,17 +95,32 @@ function handleRoundEnd(state: GameState) {
   } else if (state.ball.position.x >= 1) {
     state.playerOne.score += 1;
   }
+  // is this necessary? It is also handled in the gameloop
```
Comment (mraasvel) [inline src/backend/app/src/pong/pong.game.ts:98]: Good point, I don't think it's necessary so you can just remove it

File: src/backend/app/src/pong/pong.game.ts [hunk 2]
```diff
@@ -108,16 +129,11 @@ function updateBallPosition(state: GameState) {
...
-  ) {
-    // TODO: get the correct new direction
-    state.ball.direction.x *= -1;
-  }
+  ballBarIntersection(state.ball, state.playerOne.bar);
+  ballBarIntersection(state.ball, state.playerTwo.bar);
 }
 
-export function gameHasEnded(state: GameState) {
+export function gameHasEnded(state: GameState): boolean {
```
Comment (mraasvel) [inline src/backend/app/src/pong/pong.game.ts:136]: Maybe ping-pong type scoring: until 11 with at least 2 points difference

**Review by mraasvel — APPROVED** [review]

**Review by K1ngmar — APPROVED** [review]
👍🏽

**Pull request merged**